draft-pda-protocol.md

```diff
@@ -592,113 +564,63 @@ encoded value of the following form:
-   the task, i.e., `batch_start` and `batch_end` are multiples of
-   `PPMParam.batch_window`.
-1. The batch window of the request is at least the minimum batch window for the
-   task, i.e., `batch_end - batch_start >= PPMParam.batch_window`.
-1. The batch window of the request does not overlap with the batch window of any
-   previous request. [TODO: Enforcing this condition breaks hits, which
-   explicitly requires multiple collect requests on the same batch. We'll need
-   to fix this.]
-
-The leader responds to valid collect requests by first interacting with the
-helper as described in {{pa-aggregate}}. Once it has obtained the helper's
-encrypted output share for the batch, it responds to the collector's request
-with the following message:
+AggregateReq (Reports 1-10) -------------------------------->  \ 
+<------------------------------------ AggregateResp (State 1)  | Reports
+AggregateReq (continued, State 1)      --------------------->  | 10-11
+<------------------------------------ AggregateResp (State 2)  /
 
-~~~
-struct {
-  PPMTaskID task_id;
-  PPMProto proto;
-  PPMOutputShare leader_share;
-  opaque encrypted_helper_share;
-} PPMCollectResp;
-~~~
 
-[OPEN ISSUE: Describe how intra-protocol errors yield collect errors (see
-issue#57). For example, how does a leader respond to a collect request if the
-helper drops out?]
+AggregateReq (Reports 11-20, State 2) ---------------------->  \
+<------------------------------------ AggregateResp (State 3)  | Reports
+AggregateReq (continued, State 3) -------------------------->  | 20-21
+<------------------------------------ AggregateResp (State 4) /
 
-### Verifying and Aggregating Reports {#pa-aggregate}
+OutputShareReq (State 4) ----------------------------------->
+<----------------------------------- OutputShareResp (Result)
+~~~~
+{: #pa-aggregate-flow title="Aggregation Process (batch size=20)"}
+
+[[OPEN ISSUE: Should there be an indication of whether
+  a given AggregateReq is a continuation of a previous sub-batch]]
+[TODO: Decide if and how the collector's request is authenticated.]
 
-After the client uploads a report to the leader, the leader and helper verify in
-zero knowledge that the input is valid. The exact procedure for doing so is
-protocol specific, but all protocols have the same basic structure. In
-particular, the protocol is comprised of a sequence of *aggregate requests* from
-the leader to the helper. At the end of this procedure, the leader and helper
-will have aggregated a set of valid client inputs (though not necessarily a
-complete batch).
 
-#### Aggregate Request
+### Aggregate Request
 
-The process begins with a `PPMCollectReq`. The leader collects a sequence of
-reports that are all associated with the same PPM task. Let `[helper]` denote
+The AggregateReq request is used by the leader to send a set of reports
+to the helper. These reports MUST all be associated with the same PPM task.
+[[OPEN ISSUE: And the same batch, right?]]
```

Comment:
Yeah, that'd be pretty terrible. It seems like this would be addressed by https://github.com/abetterinternet/prio-documents/issues/109, right? Mind referring to that issue here?

Reply:
Sure.